ob1k-cache/src/main/java/com/outbrain/ob1k/cache/memcache/compression/CompressionAlgorithm.java

```diff
@@ -0,0 +1,72 @@
+package com.outbrain.ob1k.cache.memcache.compression;
+
+import com.github.luben.zstd.Zstd;
+import net.jpountz.lz4.LZ4Factory;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.function.Function;
+import java.util.zip.DataFormatException;
+import java.util.zip.Deflater;
+import java.util.zip.Inflater;
+
+/**
+ * Created by bshushi on 27/03/2017.
+ */
+public enum CompressionAlgorithm {
+  LZ4,
+  ZSTANDARD,
+  GZIP;
+
+  private static LZ4Factory lz4Factory = LZ4Factory.fastestInstance();
+
+  public Compressor getInstance() {
+    switch (this) {
+      case ZSTANDARD:
+        return new Compressor(Zstd::compress, (data) -> Zstd.decompress(data, data.length));
+      case GZIP:
+        return new Compressor(gzipCompress(), gzipDecompress());
+      case LZ4:
+        return new Compressor((data) -> lz4Factory.fastCompressor().compress(data), (data) -> lz4Factory.safeDecompressor().decompress(data, data.length));
+      default:
+        throw new RuntimeException("Unresolved compression algorithm.");
+    }
+  }
+
+  private Function<byte[], byte[]> gzipDecompress() {
```

Comment:
about the abstraction:
we can change Compressor to be interface, with implementation for each of the algorithms specified - removing all the impl related logic here (those methods, lz4 instance, etc)